File: src/ios/CDVAppDelegate+AppScope.m
/*! Copyright 2017 Ayogo Health Inc. */

#import "CDVAppDelegate+AppScope.h"

@implementation CDVAppDelegate (appScope)

- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler;
{
    if ([userActivity.activityType isEqualToString:NSUserActivityTypeBrowsingWeb]) {
        NSLog(@"APPSCOPE-PLUGIN: continueUserActivity with URL: %@", userActivity.webpageURL);

        [[NSNotificationCenter defaultCenter] postNotification:[NSNotification notificationWithName:CDVPluginHandleOpenURLNotification object:userActivity.webpageURL]];

        return YES;
    }

    return NO;
}

@end

File: src/ios/CDVAppDelegate+AppScope.swift
/*! Copyright 2017 Ayogo Health Inc. */

extension CDVAppDelegate {

    func application(_ application: UIApplication, continue userActivity:NSUserActivity, restorationHandler: @escaping ([Any]?) -> Void) -> Bool {
        NSLog("CORDOVA-PLUGIN: Called continueUserActivity")

        if userActivity.activityType == NSUserActivityTypeBrowsingWeb {
            return true
        }

        return false
    }
}
